anti-tracking-threat-model.md

```diff
@@ -0,0 +1,196 @@
+with proposed or already-adopted mitigations to frustrate those goals. Not all
+of the attacks use Web Packaging, so that we can explore the threat model's
+implications for the rest of the web platform.
+
+## Sign identifying information into the package
+
+### The Attack
+
+1. AdTech convinces News to give AdTech a package-signing certificate for
+   `news.example`, perhaps by offering to handle the technical complications of
+   signing packages.
+   1. Alternately, News could set up a service that signs packages AdTech sends
+      it.
+1. When a user clicks a link from `adtech.example` to
+   `https://adtech.example/news.example.sxg`, they send identifying information
+   to the `adtech.example` server. This could be cookies or a user ID encoded in
+   the query, path, or even hostname.
+1. Instead of signing `news.example`'s content directly, AdTech embeds the
+   user's identity in that content and signs the result on the fly.
+1. This successfully transfers the unique ID to the `news.example` JS
+   environment, where it can be picked up by advertising code there.
+
+### Proposed Mitigations
+
+#### Preflight to publisher
+
+1. The server responding with the signed package is required to send the
+   signature up front. This does not prevent any attacks but increases the
+   user-visible latency, which AdTech experiences as a cost.
+2. The user agent makes an ephemeral, cookie-less preflight request to
+   `news.example` to get the signature and then validates the package from
+   `adtech.example` against that signature.
+
+   Both the
+   [fully-offline](https://wicg.github.io/webpackage/draft-yasskin-webpackage-use-cases.html#fully-offline-use)
+   use case and [cryptographic
+   agility](https://wicg.github.io/webpackage/draft-yasskin-webpackage-use-cases.html#crypto-agility)
+   require `news.example` to be able to declare that multiple signatures are
+   currently valid. This requires AdTech to send all of its signatures to
+   `news.example` and for News to list them all in the now-quite-large preflight
+   response. Having enough signatures that each one identifies a user would be
+   detectable by clients and might be enough cost that News wouldn't be willing
+   to do it.
+3. We add a signed time stamp to the package signature to avoid AdTech telling
+   News to get signatures from `adtech.example` backend and send personalized
+   signatures back as preflight responses. With such time stamps, the user agent
+   can decide to not accept signatures younger than, say one minute. For this to
+   work we need signed, official time.
+
+   TODO: Figure out how this blocks the attack.
+
+#### Public signature repository
+
+Another potential mitigation would be some kind of public repository of
+signatures to check against.
+
+This does not prevent any attacks, but could make them detectable.
+
+#### Make package loads stateless
+
+When requesting a signed package:
```

Comment:
I've only given casual thought to how a webpage can instruct the browser to restrict a load because it is for a signed package. Two ideas: 1) a /.well-known/ location or 2) a special HTTP redirect similar to an upgrade.